modules/graphics/src/mrb_vertex_buffer.cxx

```diff
@@ -95,6 +99,50 @@ vbo_index_count(mrb_state *mrb, mrb_value self)
   return mrb_fixnum_value(mmrb_vertex_buffer_ptr(mrb, self)->GetIndexCount());
 }
 
+static mrb_value
+vbo_add_quad(mrb_state *mrb, mrb_value self)
+{
+  Moon::IntRect quad_rect;
+  Moon::FloatRect quad_texture_rect;
+  Moon::Vector4 color;
+  Moon::VertexBuffer *vbo = NULL;
+  Moon::Vertex vertices[4];
+  GLuint indices[] = { 0, 1, 3, 2, 3, 1 };
+
+  mrb_value rect_obj;
+  mrb_value tex_rect_obj;
+  mrb_value color_obj;
+  mrb_int i = 0;
+
+  mrb_get_args(mrb, "ooo", &rect_obj, &tex_rect_obj, &color_obj);
+  quad_rect = mmrb_to_rect(mrb, rect_obj);
+  quad_texture_rect = mmrb_to_float_rect(mrb, tex_rect_obj);
+  color = mmrb_to_vector4(mrb, tex_rect_obj);
+  vbo = mmrb_vertex_buffer_ptr(mrb, self);
+
+  vertices[i].pos = Moon::Vector2(quad_rect.x, quad_rect.y);
```

Comment:
Instead of initializing and then doing assignments, I prefer to do object initialization:

```cpp
Moon::Vertex vertices[4] = {
  { {quad_rect.x, quad_rect.y}, {quad_texture_rect.x, quad_texture_rect.y}, color },
  //...
};
```